File: family_finance_api/FamilyFinance/Models/Entities/Account.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace FamilyFinance.Models.Entities;

[Table("accounts")]
public class Account : BaseModel<int>
{
    [Column("name")]
    [MaxLength(100)]
    public string Name { get; set; } = string.Empty;

    [Column("description")]
    [MaxLength(256)]
    public string Description { get; set; } = string.Empty;

    [Column("color")]
    [MaxLength(7)]
    public string Color { get; set; } = string.Empty;

    [Column("balance")]
    public decimal Balance { get; set; }

    [Column("user_id")]
    public int UserId { get; set; }

    [ForeignKey("UserId")]
    public virtual User User { get; set; } = null!;

    public virtual ICollection<Transaction> Transactions { get; set; } = [];
}

File: family_finance_api_go/internal/models/account.go
package models

import (
	"time"

	"github.com/dimasyanu/family-finance-go/internal/models/request"
	"gorm.io/gorm"
)

type Account struct {
	Name        string  `gorm:"column:name;type:varchar(100);not null"`
	Description string  `gorm:"column:description;type:text"`
	Color       string  `gorm:"column:color;type:varchar(20)"`
	Balance     float64 `gorm:"column:balance;type:decimal(18,2);not null;default:0"`
	UserID      uint    `gorm:"column:user_id;type:int;not null;index"`

	User *User `gorm:"foreignKey:UserID;references:ID"`

	gorm.Model
}

func FromCreateAccountRequest(payload *request.SaveAccountRequest) *Account {
	return &Account{
		UserID:      payload.UserID,
		Name:        payload.Name,
		Description: payload.Description,
		Color:       payload.Color,
		Balance:     payload.Balance,

		Model: gorm.Model{CreatedAt: time.Now(), UpdatedAt: time.Now()},
	}
}
